tests/unit/lib/observers/ParameterObserverTest_unittest.cc

```diff
@@ -41,7 +46,18 @@ class MockEmitter : public SGObject
 		observe<int32_t>(1, "a", 1, p);
 		observe<int32_t>(1, "b", 1, p2);
 		observe<int32_t>(1, "None", 1, p3);
+
+		std::function<int32_t()> lambda_function = [this]() {
+			this->member_variable++;
+			return 20;
+		};
+
+		observe<std::function<int32_t()>>(
```

Comment:
Apparently, `Any` is smarter than I thought :) with the current implementation we can emit `std::function` directly without changing anything in the framework. We can also execute them in a fairly simple way by just calling `get<lambda return type>("name of the lambda")`. See the unit tests for an example.

Next question. Some of these operations could be quite costly and if the user calls them multiple times then they will be executed from scratch every time. Shall we somehow cache the result of these executions, so that the user will just need to call the lambda once and then it will retrieve only the pre-computed result?

This caching will happen only for the stored observations though.

Reply:
could you make an example of such an operation?

Reply:
> could you make an example of such an operation?

mmh I don't have any real example now. I was thinking about possible use cases/corner cases that we may face.

Reply:
@geektoni I am modifying things a bit in #5037, so we use visitor patterns instead of `any_cast`. The only things you will have to keep in mind is that the visitor will only call a function when you get it if you have the `ParameterProperties::FUNCTION`, which is added in that PR. About the caching, you could use a `static` variable inside the lambda.

Reply:
actually, maybe don't use static, that might cause massive issues

Reply:
If you want to cache actually, use `std::optional`. Put the result there. If it is set it returns the value, otherwise computes it.